### PR TITLE
Adjusts in the package accounts-2fa

### PR DESCRIPTION
**Pull request by denihs**
- Throwing error when a user is not found on `Accounts._is2faEnabledForUser`
- Remove vulnerability from the method `has2faEnabled`. It no longer takes a selector as an argument, just the callback.
- Auto-publishing the field `services.twoFactorAuthentication.type` for logged users.
- `generate2faActivationQrCode` now throws an error if it's being called when the user already has 2FA enabled.

### CODE REVIEW

File: packages/accounts-2fa/2fa-server.js [hunk 1]
```diff
@@ -29,7 +29,13 @@ Accounts._is2faEnabledForUser = selector => {
     }
   }
 
-  const user = Meteor.users.findOne(selector) || {};
+  const user = Meteor.users.findOne(selector);
+  if (!user) {
+    throw new Meteor.Error(
+      500,
```
Comment (StorytellerCZ) [inline packages/accounts-2fa/2fa-server.js:35]: Maybe 404 error would be more appropriate here?

**Review by StorytellerCZ — CHANGES_REQUESTED** [review]

File: packages/accounts-2fa/2fa-server.js (hunk 1 above)
Comment (fredmaiaarantes) [reply packages/accounts-2fa/2fa-server.js:35]: I think it makes more sense. The docs also need to be updated to 404 in that case.

Comment (harryadel) [reply packages/accounts-2fa/2fa-server.js:35]: Sorry to barge in, but why use HTTP error codes?? This is DDP, won't a descriptive title be better? 
https://github.com/meteor/meteor/blob/365604c91020db2c36b491e7ea332ed730492994/packages/ddp-server/livedata_server.js#L663

Comment (denihs) [reply packages/accounts-2fa/2fa-server.js:35]: It's the pattern that the packages are using, but sure enough, I'm going to use a descriptive code in this case.

File: docs/history.md [hunk 2]
```diff
@@ -4,7 +4,8 @@
 
 #### Breaking Changes
 
-N/A
+* `accounts-2fa@1.0.1`
```
Comment (zodern) [inline docs/history.md:7]: Since there is a breaking change, the package's major version should be increased.

**Pull request merged**